ckks/bootstrapping/bootstrapping.go

```diff
@@ -8,47 +8,75 @@ import (
 	"github.com/tuneinsight/lattigo/v4/rlwe"
 )
 
-// Bootstrap re-encrypts a ciphertext at lvl Q0 to a ciphertext at MaxLevel-k where k is the depth of the bootstrapping circuit.
+// Bootstrapp re-encrypt a ciphertext at lvl Q0 to a ciphertext at MaxLevel-k where k is the depth of the bootstrapping circuit.
 // If the input ciphertext level is zero, the input scale must be an exact power of two smaller or equal to round(Q0/2^{10}).
 // If the input ciphertext is at level one or more, the input scale does not need to be an exact power of two as one level
 // can be used to do a scale matching.
-func (btp *Bootstrapper) Bootstrap(ctIn *rlwe.Ciphertext) (ctOut *rlwe.Ciphertext) {
+func (btp *Bootstrapper) Bootstrapp(ctIn *rlwe.Ciphertext) (ctOut *rlwe.Ciphertext) {
 
```

Comment:
In `examples/ckks/bootstrapping/main.go`, there is a comment:
```
// CAUTION: the scale of the ciphertext MUST be equal (or very close) to params.DefaultScale()
```
I _strongly_ advise that you assert this condition here explicitly (panicking if the condition is not met), at the top of `Bootstrapp()` and remove the CAUTION comment. I lost ~3 days tracing down a bug that would have been obvious if there was an assertion here.

Reply:
Thank you for pointing it out, I added this check.